diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/auto/AutoBuilder.java b/pathplannerlib/src/main/java/com/pathplanner/lib/auto/AutoBuilder.java
index cc5fdf2..114af72 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/auto/AutoBuilder.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/auto/AutoBuilder.java
@@ -21,7 +21,7 @@ import org.wpilib.command2.Subsystem;
 import org.wpilib.driverstation.DriverStationErrors;
 import org.wpilib.math.geometry.Pose2d;
 import org.wpilib.math.kinematics.ChassisVelocities;
-import org.wpilib.smartdashboard.SendableChooser;
+import org.wpilib.tunable.Selectable;
 import org.wpilib.system.Filesystem;
 import org.wpilib.units.measure.LinearVelocity;
 
@@ -394,7 +394,7 @@ public class AutoBuilder {
    *
    * @return SendableChooser populated with all autos
    */
-  public static SendableChooser<Command> buildAutoChooser() {
+  public static Selectable<Command> buildAutoChooser() {
     return buildAutoChooser("");
   }
 
@@ -406,7 +406,7 @@ public class AutoBuilder {
    *     Commands.none()
    * @return SendableChooser populated with all autos
    */
-  public static SendableChooser<Command> buildAutoChooser(String defaultAutoName) {
+  public static Selectable<Command> buildAutoChooser(String defaultAutoName) {
     return buildAutoChooserWithOptionsModifier(defaultAutoName, (stream) -> stream);
   }
 
@@ -418,7 +418,7 @@ public class AutoBuilder {
    *     into the AutoChooser
    * @return SendableChooser populated with all autos
    */
-  public static SendableChooser<Command> buildAutoChooserWithOptionsModifier(
+  public static Selectable<Command> buildAutoChooserWithOptionsModifier(
       Function<Stream<PathPlannerAuto>, Stream<PathPlannerAuto>> optionsModifier) {
     return buildAutoChooserWithOptionsModifier("", optionsModifier);
   }
@@ -433,14 +433,14 @@ public class AutoBuilder {
    *     into the AutoChooser
    * @return SendableChooser populated with all autos
    */
-  public static SendableChooser<Command> buildAutoChooserWithOptionsModifier(
+  public static Selectable<Command> buildAutoChooserWithOptionsModifier(
       String defaultAutoName,
       Function<Stream<PathPlannerAuto>, Stream<PathPlannerAuto>> optionsModifier) {
     if (!AutoBuilder.isConfigured()) {
       throw new RuntimeException(
           "AutoBuilder was not configured before attempting to build an auto chooser");
     }
-    SendableChooser<Command> chooser = new SendableChooser<>();
+    Selectable<Command> chooser = new Selectable<>();
     List<String> autoNames = getAllAutoNames();
     PathPlannerAuto defaultOption = null;
     List<PathPlannerAuto> options = new ArrayList<>();
@@ -453,14 +453,14 @@ public class AutoBuilder {
       }
     }
     if (defaultOption == null) {
-      chooser.setDefaultOption("None", Commands.none());
+      chooser.addDefault("None", Commands.none());
     } else {
-      chooser.setDefaultOption(defaultOption.getName(), defaultOption);
-      chooser.addOption("None", Commands.none());
+      chooser.addDefault(defaultOption.getName(), defaultOption);
+      chooser.add("None", Commands.none());
     }
     optionsModifier
         .apply(options.stream())
-        .forEach(auto -> chooser.addOption(auto.getName(), auto));
+        .forEach(auto -> chooser.add(auto.getName(), auto));
     return chooser;
   }
 
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/auto/NamedCommands.java b/pathplannerlib/src/main/java/com/pathplanner/lib/auto/NamedCommands.java
index 6b15ec8..994fbed 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/auto/NamedCommands.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/auto/NamedCommands.java
@@ -5,7 +5,7 @@ import java.util.List;
 import java.util.Map;
 import org.wpilib.command2.*;
 import org.wpilib.driverstation.DriverStationErrors;
-import org.wpilib.math.util.Pair;
+import org.wpilib.util.Pair;
 
 /** Utility class for managing named commands */
 public class NamedCommands {
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/FollowPathCommand.java b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/FollowPathCommand.java
index 2899df0..2586cdb 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/FollowPathCommand.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/FollowPathCommand.java
@@ -192,16 +192,16 @@ public class FollowPathCommand extends Command {
   public static Command warmupCommand() {
     List<Waypoint> waypoints =
         PathPlannerPath.waypointsFromPoses(
-            new Pose2d(0.0, 0.0, Rotation2d.kZero), new Pose2d(6.0, 6.0, Rotation2d.kZero));
+            new Pose2d(0.0, 0.0, Rotation2d.ZERO), new Pose2d(6.0, 6.0, Rotation2d.ZERO));
     PathPlannerPath path =
         new PathPlannerPath(
             waypoints,
             new PathConstraints(4.0, 4.0, 4.0, 4.0),
-            new IdealStartingState(0.0, Rotation2d.kZero),
-            new GoalEndState(0.0, Rotation2d.kCW_90deg));
+            new IdealStartingState(0.0, Rotation2d.ZERO),
+            new GoalEndState(0.0, Rotation2d.CW_90DEG));
     return new FollowPathCommand(
             path,
-            () -> Pose2d.kZero,
+            () -> Pose2d.ZERO,
             ChassisVelocities::new,
             (speeds, feedforwards) -> {},
             new PPHolonomicDriveController(
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathPlannerAuto.java b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathPlannerAuto.java
index 7b0ccba..325b0de 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathPlannerAuto.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathPlannerAuto.java
@@ -160,7 +160,7 @@ public class PathPlannerAuto extends Command {
    * @param autoCommand The command this auto should run
    */
   public PathPlannerAuto(Command autoCommand) {
-    this(autoCommand, Pose2d.kZero);
+    this(autoCommand, Pose2d.ZERO);
   }
 
   /**
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathfindingCommand.java b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathfindingCommand.java
index 4c7a17b..1c2d0f9 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathfindingCommand.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/commands/PathfindingCommand.java
@@ -93,7 +93,7 @@ public class PathfindingCommand extends Command {
       Subsystem... requirements) {
     addRequirements(requirements);
     Pathfinding.ensureInitialized();
-    Rotation2d targetRotation = Rotation2d.kZero;
+    Rotation2d targetRotation = Rotation2d.ZERO;
     double goalEndVel = targetPath.getGlobalConstraints().maxVelocityMPS();
     if (targetPath.isChoreoPath()) {
       // Can get() here without issue since all choreo trajectories have ideal trajectories
@@ -396,9 +396,9 @@ public class PathfindingCommand extends Command {
    */
   public static Command warmupCommand() {
     return new PathfindingCommand(
-            new Pose2d(15.0, 4.0, Rotation2d.k180deg),
+            new Pose2d(15.0, 4.0, Rotation2d.PI),
             new PathConstraints(4, 3, 4, 4),
-            () -> new Pose2d(1.5, 4, Rotation2d.kZero),
+            () -> new Pose2d(1.5, 4, Rotation2d.ZERO),
             ChassisVelocities::new,
             (speeds, feedforwards) -> {},
             new PPHolonomicDriveController(
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/config/RobotConfig.java b/pathplannerlib/src/main/java/com/pathplanner/lib/config/RobotConfig.java
index dfdb9dd..9a16213 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/config/RobotConfig.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/config/RobotConfig.java
@@ -7,8 +7,8 @@ import org.ejml.simple.SimpleMatrix;
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
-import org.wpilib.driverstation.Alert;
-import org.wpilib.driverstation.Alert.Level;
+import org.wpilib.util.Alert;
+import org.wpilib.util.Alert.Level;
 import org.wpilib.math.geometry.Rotation2d;
 import org.wpilib.math.geometry.Translation2d;
 import org.wpilib.math.kinematics.*;
@@ -60,33 +60,33 @@ public class RobotConfig {
 
   // Validation alerts
   private static final Alert BAD_GUI_CONFIG =
-      new Alert("PathPlanner", "GUI Config Couldn't be loaded", Level.HIGH);
+      new Alert("PathPlanner", "GUI Config Couldn't be loaded", "GUI Config Couldn't be loaded", Level.HIGH);
 
   private static final Alert MOI_ALERT =
-      new Alert("PathPlanner", "MOI Config Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "MOI Config Mismatch", "MOI Config Mismatch", Level.HIGH);
 
   private static final Alert MASS_ALERT =
-      new Alert("PathPlanner", "Mass Config Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Mass Config Mismatch", "Mass Config Mismatch", Level.HIGH);
 
   private static final Alert TORQUE_ALERT =
-      new Alert("PathPlanner", "Torque Friction Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Torque Friction Mismatch", "Torque Friction Mismatch", Level.HIGH);
 
   private static final Alert CURRENT_ALERT =
-      new Alert("PathPlanner", "Drive Current Limit Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Drive Current Limit Mismatch", "Drive Current Limit Mismatch", Level.HIGH);
 
   private static final Alert MOTOR_ALERT =
-      new Alert("PathPlanner", "Drive Motor Config Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Drive Motor Config Mismatch", "Drive Motor Config Mismatch", Level.HIGH);
 
   private static final Alert VELOCITY_ALERT =
-      new Alert("PathPlanner", "Max Drive Velocity Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Max Drive Velocity Mismatch", "Max Drive Velocity Mismatch", Level.HIGH);
 
-  private static final Alert COF_ALERT = new Alert("PathPlanner", "Wheel COF Mismatch", Level.HIGH);
+  private static final Alert COF_ALERT = new Alert("PathPlanner", "Wheel COF Mismatch", "Wheel COF Mismatch", Level.HIGH);
 
   private static final Alert RADIUS_ALERT =
-      new Alert("PathPlanner", "Wheel Radius Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Wheel Radius Mismatch", "Wheel Radius Mismatch", Level.HIGH);
 
   private static final Alert LOCATION_ALERT =
-      new Alert("PathPlanner", "Module Location Mismatch", Level.HIGH);
+      new Alert("PathPlanner", "Module Location Mismatch", "Module Location Mismatch", Level.HIGH);
 
   /**
    * Create a robot config object for a HOLONOMIC DRIVE robot
@@ -121,7 +121,7 @@ public class RobotConfig {
     for (int i = 0; i < this.numModules; i++) {
       Translation2d modPosReciprocal =
           new Translation2d(
-              1.0 / this.moduleLocations[i].getNorm(), this.moduleLocations[i].getAngle());
+              1.0 / this.moduleLocations[i].getNorm(), this.moduleLocations[i].getAngle().orElse(Rotation2d.ZERO));
       this.forceKinematics.setRow(i * 2, 0, /* Start Data */ 1, 0, -modPosReciprocal.getY());
       this.forceKinematics.setRow(i * 2 + 1, 0, /* Start Data */ 0, 1, modPosReciprocal.getX());
     }
@@ -174,7 +174,7 @@ public class RobotConfig {
     for (int i = 0; i < this.numModules; i++) {
       Translation2d modPosReciprocal =
           new Translation2d(
-              1.0 / this.moduleLocations[i].getNorm(), this.moduleLocations[i].getAngle());
+              1.0 / this.moduleLocations[i].getNorm(), this.moduleLocations[i].getAngle().orElse(Rotation2d.ZERO));
       this.forceKinematics.setRow(i * 2, 0, /* Start Data */ 1, 0, -modPosReciprocal.getY());
       this.forceKinematics.setRow(i * 2 + 1, 0, /* Start Data */ 0, 1, modPosReciprocal.getX());
     }
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/path/PathPlannerPath.java b/pathplannerlib/src/main/java/com/pathplanner/lib/path/PathPlannerPath.java
index 2ceebc5..5d019a7 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/path/PathPlannerPath.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/path/PathPlannerPath.java
@@ -657,7 +657,7 @@ public class PathPlannerPath {
    * @return Initial heading
    */
   public Rotation2d getInitialHeading() {
-    return getPoint(1).position.minus(getPoint(0).position).getAngle();
+    return getPoint(1).position.minus(getPoint(0).position).getAngle().orElse(Rotation2d.ZERO);
   }
 
   /**
@@ -846,7 +846,7 @@ public class PathPlannerPath {
       var pointZone = pointZoneForWaypointPos(points.get(i).waypointRelativePos);
       if (pointZone != null) {
         Rotation2d angleToTarget =
-            pointZone.targetPosition().minus(points.get(i).position).getAngle();
+            pointZone.targetPosition().minus(points.get(i).position).getAngle().orElse(Rotation2d.ZERO);
         Rotation2d rotation = angleToTarget.plus(pointZone.rotationOffset());
         points.get(i).rotationTarget =
             new RotationTarget(points.get(i).waypointRelativePos, rotation);
@@ -1306,7 +1306,7 @@ public class PathPlannerPath {
    */
   public List<Pose2d> getPathPoses() {
     return allPoints.stream()
-        .map(p -> new Pose2d(p.position, Rotation2d.kZero))
+        .map(p -> new Pose2d(p.position, Rotation2d.ZERO))
         .collect(Collectors.toList());
   }
 
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/path/PointTowardsZone.java b/pathplannerlib/src/main/java/com/pathplanner/lib/path/PointTowardsZone.java
index 4399bb4..8367d27 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/path/PointTowardsZone.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/path/PointTowardsZone.java
@@ -37,7 +37,7 @@ public record PointTowardsZone(
       Translation2d targetPosition,
       double minWaypointRelativePos,
       double maxWaypointRelativePos) {
-    this(name, targetPosition, Rotation2d.kZero, minWaypointRelativePos, maxWaypointRelativePos);
+    this(name, targetPosition, Rotation2d.ZERO, minWaypointRelativePos, maxWaypointRelativePos);
   }
 
   /**
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/LocalADStar.java b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/LocalADStar.java
index ceaf96e..3074bfa 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/LocalADStar.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/LocalADStar.java
@@ -13,7 +13,7 @@ import org.json.simple.parser.JSONParser;
 import org.wpilib.math.geometry.Pose2d;
 import org.wpilib.math.geometry.Rotation2d;
 import org.wpilib.math.geometry.Translation2d;
-import org.wpilib.math.util.Pair;
+import org.wpilib.util.Pair;
 import org.wpilib.system.Filesystem;
 
 /**
@@ -86,9 +86,9 @@ public class LocalADStar implements Pathfinder {
   public LocalADStar() {
     planningThread = new Thread(this::runThread);
     requestStart = new GridPosition(0, 0);
-    requestRealStartPos = Translation2d.kZero;
+    requestRealStartPos = Translation2d.ZERO;
     requestGoal = new GridPosition(0, 0);
-    requestRealGoalPos = Translation2d.kZero;
+    requestRealGoalPos = Translation2d.ZERO;
     staticObstacles.clear();
     dynamicObstacles.clear();
     File navGridFile = new File(Filesystem.getDeployDirectory(), "pathplanner/navgrid.json");
@@ -391,15 +391,15 @@ public class LocalADStar implements Pathfinder {
     fieldPosPath.set(fieldPosPath.size() - 1, realGoalPos);
     List<Pose2d> pathPoses = new ArrayList<>();
     pathPoses.add(
-        new Pose2d(fieldPosPath.get(0), fieldPosPath.get(1).minus(fieldPosPath.get(0)).getAngle()));
+        new Pose2d(fieldPosPath.get(0), fieldPosPath.get(1).minus(fieldPosPath.get(0)).getAngle().orElse(Rotation2d.ZERO)));
     for (int i = 1; i < fieldPosPath.size() - 1; i++) {
       Translation2d last = fieldPosPath.get(i - 1);
       Translation2d current = fieldPosPath.get(i);
       Translation2d next = fieldPosPath.get(i + 1);
       Translation2d anchor1 = current.minus(last).times(SMOOTHING_ANCHOR_PCT).plus(last);
-      Rotation2d heading1 = current.minus(last).getAngle();
+      Rotation2d heading1 = current.minus(last).getAngle().orElse(Rotation2d.ZERO);
       Translation2d anchor2 = current.minus(next).times(SMOOTHING_ANCHOR_PCT).plus(next);
-      Rotation2d heading2 = next.minus(anchor2).getAngle();
+      Rotation2d heading2 = next.minus(anchor2).getAngle().orElse(Rotation2d.ZERO);
       pathPoses.add(new Pose2d(anchor1, heading1));
       pathPoses.add(new Pose2d(anchor2, heading2));
     }
@@ -409,7 +409,7 @@ public class LocalADStar implements Pathfinder {
             fieldPosPath
                 .get(fieldPosPath.size() - 1)
                 .minus(fieldPosPath.get(fieldPosPath.size() - 2))
-                .getAngle()));
+                .getAngle().orElse(Rotation2d.ZERO)));
     return PathPlannerPath.waypointsFromPoses(pathPoses);
   }
 
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinder.java b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinder.java
index 879f2fa..2689df3 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinder.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinder.java
@@ -5,7 +5,7 @@ import com.pathplanner.lib.path.PathConstraints;
 import com.pathplanner.lib.path.PathPlannerPath;
 import java.util.List;
 import org.wpilib.math.geometry.Translation2d;
-import org.wpilib.math.util.Pair;
+import org.wpilib.util.Pair;
 
 /** Interface for a pathfinder that can be used by PPLib's pathfinding commands */
 public interface Pathfinder {
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinding.java b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinding.java
index 32b9625..d594e9a 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinding.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/pathfinding/Pathfinding.java
@@ -5,7 +5,7 @@ import com.pathplanner.lib.path.PathConstraints;
 import com.pathplanner.lib.path.PathPlannerPath;
 import java.util.List;
 import org.wpilib.math.geometry.Translation2d;
-import org.wpilib.math.util.Pair;
+import org.wpilib.util.Pair;
 
 /**
  * Static class for interacting with the chosen pathfinding implementation from the pathfinding
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectory.java b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectory.java
index 206d23a..4e1f090 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectory.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectory.java
@@ -167,7 +167,7 @@ public class PathPlannerTrajectory {
             double appliedForce =
                 wheelForceDist > 1e-6
                     ? wheelForceDist
-                        * wheelForces[m].getAngle().minus(state.moduleStates[m].angle).getCos()
+                        * wheelForces[m].getAngle().orElse(Rotation2d.ZERO).minus(state.moduleStates[m].angle).getCos()
                     : 0.0;
             double wheelTorque = appliedForce * config.moduleConfig.wheelRadiusMeters;
             double torqueCurrent = config.moduleConfig.driveMotor.getCurrent(wheelTorque);
@@ -243,9 +243,9 @@ public class PathPlannerTrajectory {
         Translation2d headingTranslation =
             path.getPoint(i + 1).position.minus(state.pose.getTranslation());
         if (headingTranslation.getNorm() <= 1e-6) {
-          state.heading = Rotation2d.kZero;
+          state.heading = Rotation2d.ZERO;
         } else {
-          state.heading = headingTranslation.getAngle();
+          state.heading = headingTranslation.getAngle().orElse(Rotation2d.ZERO);
         }
       } else {
         state.heading = states.get(i - 1).heading;
@@ -285,9 +285,9 @@ public class PathPlannerTrajectory {
                   .fieldPos
                   .minus(states.get(i).moduleStates[m].fieldPos);
           if (fieldTranslation.getNorm() <= 1e-6) {
-            states.get(i).moduleStates[m].fieldAngle = Rotation2d.kZero;
+            states.get(i).moduleStates[m].fieldAngle = Rotation2d.ZERO;
           } else {
-            states.get(i).moduleStates[m].fieldAngle = fieldTranslation.getAngle();
+            states.get(i).moduleStates[m].fieldAngle = fieldTranslation.getAngle().orElse(Rotation2d.ZERO);
           }
           states.get(i).moduleStates[m].angle =
               states.get(i).moduleStates[m].fieldAngle.minus(states.get(i).pose.getRotation());
@@ -308,7 +308,7 @@ public class PathPlannerTrajectory {
       var nextState = states.get(i + 1);
 
       // Calculate the linear force vector and torque acting on the whole robot
-      Translation2d linearForceVec = Translation2d.kZero;
+      Translation2d linearForceVec = Translation2d.ZERO;
       double totalTorque = 0.0;
       for (int m = 0; m < config.numModules; m++) {
         double lastVel = prevState.moduleStates[m].velocity;
@@ -332,12 +332,12 @@ public class PathPlannerTrajectory {
 
         // Calculate the torque this module will apply to the robot
         Rotation2d angleToModule =
-            state.moduleStates[m].fieldPos.minus(state.pose.getTranslation()).getAngle();
+            state.moduleStates[m].fieldPos.minus(state.pose.getTranslation()).getAngle().orElse(Rotation2d.ZERO);
         Rotation2d theta;
         if (forceVec.getNorm() <= 1e-6) {
-          theta = Rotation2d.kZero.minus(angleToModule);
+          theta = Rotation2d.ZERO.minus(angleToModule);
         } else {
-          theta = forceVec.getAngle().minus(angleToModule);
+          theta = forceVec.getAngle().orElse(Rotation2d.ZERO).minus(angleToModule);
         }
         totalTorque += forceAtCarpet * config.modulePivotDistance[m] * theta.getSin();
       }
@@ -448,7 +448,7 @@ public class PathPlannerTrajectory {
       var nextState = states.get(i + 1);
 
       // Calculate the linear force vector and torque acting on the whole robot
-      Translation2d linearForceVec = Translation2d.kZero;
+      Translation2d linearForceVec = Translation2d.ZERO;
       double totalTorque = 0.0;
       for (int m = 0; m < config.numModules; m++) {
         double lastVel = nextState.moduleStates[m].velocity;
@@ -466,19 +466,19 @@ public class PathPlannerTrajectory {
 
         Translation2d forceVec =
             new Translation2d(
-                forceAtCarpet, state.moduleStates[m].fieldAngle.plus(Rotation2d.k180deg));
+                forceAtCarpet, state.moduleStates[m].fieldAngle.plus(Rotation2d.PI));
 
         // Add the module force vector to the robot force vector
         linearForceVec = linearForceVec.plus(forceVec);
 
         // Calculate the torque this module will apply to the robot
         Rotation2d angleToModule =
-            state.moduleStates[m].fieldPos.minus(state.pose.getTranslation()).getAngle();
+            state.moduleStates[m].fieldPos.minus(state.pose.getTranslation()).getAngle().orElse(Rotation2d.ZERO);
         Rotation2d theta;
         if (forceVec.getNorm() <= 1e-6) {
-          theta = Rotation2d.kZero.minus(angleToModule);
+          theta = Rotation2d.ZERO.minus(angleToModule);
         } else {
-          theta = forceVec.getAngle().minus(angleToModule);
+          theta = forceVec.getAngle().orElse(Rotation2d.ZERO).minus(angleToModule);
         }
         totalTorque += forceAtCarpet * config.modulePivotDistance[m] * theta.getSin();
       }
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectoryState.java b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectoryState.java
index f667d63..e5c2c8f 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectoryState.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/PathPlannerTrajectoryState.java
@@ -20,13 +20,13 @@ public class PathPlannerTrajectoryState implements Interpolatable<PathPlannerTra
   public ChassisVelocities fieldSpeeds = new ChassisVelocities();
 
   /** Field-relative robot pose at this state */
-  public Pose2d pose = Pose2d.kZero;
+  public Pose2d pose = Pose2d.ZERO;
 
   /** The linear velocity at this state in m/s */
   public double linearVelocity = 0.0;
 
   /** The field-relative heading, or direction of travel, at this state */
-  public Rotation2d heading = Rotation2d.kZero;
+  public Rotation2d heading = Rotation2d.ZERO;
 
   /** The feedforwards for each module */
   public DriveFeedforwards feedforwards;
@@ -36,7 +36,7 @@ public class PathPlannerTrajectoryState implements Interpolatable<PathPlannerTra
   protected double deltaPos = 0.0;
 
   /** The difference in rotation between this state and the previous state */
-  protected Rotation2d deltaRot = Rotation2d.kZero;
+  protected Rotation2d deltaRot = Rotation2d.ZERO;
 
   /**
    * The {@link com.pathplanner.lib.trajectory.SwerveModuleTrajectoryState} states for this state
@@ -117,13 +117,13 @@ public class PathPlannerTrajectoryState implements Interpolatable<PathPlannerTra
 
     reversed.timeSeconds = timeSeconds;
     Translation2d reversedSpeeds =
-        new Translation2d(fieldSpeeds.vx, fieldSpeeds.vy).rotateBy(Rotation2d.k180deg);
+        new Translation2d(fieldSpeeds.vx, fieldSpeeds.vy).rotateBy(Rotation2d.PI);
     reversed.fieldSpeeds =
         new ChassisVelocities(reversedSpeeds.getX(), reversedSpeeds.getY(), fieldSpeeds.omega);
-    reversed.pose = new Pose2d(pose.getTranslation(), pose.getRotation().plus(Rotation2d.k180deg));
+    reversed.pose = new Pose2d(pose.getTranslation(), pose.getRotation().plus(Rotation2d.PI));
     reversed.linearVelocity = -linearVelocity;
     reversed.feedforwards = feedforwards.reverse();
-    reversed.heading = heading.plus(Rotation2d.k180deg);
+    reversed.heading = heading.plus(Rotation2d.PI);
 
     return reversed;
   }
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/SwerveModuleTrajectoryState.java b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/SwerveModuleTrajectoryState.java
index 5fdac74..a55d05c 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/SwerveModuleTrajectoryState.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/trajectory/SwerveModuleTrajectoryState.java
@@ -8,10 +8,10 @@ import org.wpilib.math.kinematics.SwerveModuleVelocity;
 public class SwerveModuleTrajectoryState extends SwerveModuleVelocity {
 
   /** Field relative angle of the swerve module */
-  protected Rotation2d fieldAngle = Rotation2d.kZero;
+  protected Rotation2d fieldAngle = Rotation2d.ZERO;
 
   /** Position of this module on the field */
-  protected Translation2d fieldPos = Translation2d.kZero;
+  protected Translation2d fieldPos = Translation2d.ZERO;
 
   /** Difference in module position between this state and the previous state */
   protected double deltaPos = 0.0;
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/util/FlippingUtil.java b/pathplannerlib/src/main/java/com/pathplanner/lib/util/FlippingUtil.java
index 0cddfb7..e36ab69 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/util/FlippingUtil.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/util/FlippingUtil.java
@@ -50,8 +50,8 @@ public class FlippingUtil {
    */
   public static Rotation2d flipFieldRotation(Rotation2d rotation) {
     return switch (symmetryType) {
-      case kMirrored -> Rotation2d.kPi.minus(rotation);
-      case kRotational -> rotation.minus(Rotation2d.kPi);
+      case kMirrored -> Rotation2d.PI.minus(rotation);
+      case kRotational -> rotation.minus(Rotation2d.PI);
     };
   }
 
diff --git a/pathplannerlib/src/main/java/com/pathplanner/lib/util/swerve/SwerveSetpointGenerator.java b/pathplannerlib/src/main/java/com/pathplanner/lib/util/swerve/SwerveSetpointGenerator.java
index fe7f70c..7498320 100644
--- a/pathplannerlib/src/main/java/com/pathplanner/lib/util/swerve/SwerveSetpointGenerator.java
+++ b/pathplannerlib/src/main/java/com/pathplanner/lib/util/swerve/SwerveSetpointGenerator.java
@@ -113,7 +113,8 @@ public class SwerveSetpointGenerator {
     SwerveModuleVelocity[] desiredModuleStates =
         config.toSwerveModuleStates(desiredStateRobotRelative);
     // Make sure desiredState respects velocity limits.
-    SwerveDriveKinematics.desaturateWheelVelocities(desiredModuleStates, maxSpeed);
+    desiredModuleStates =
+        SwerveDriveKinematics.desaturateWheelVelocities(desiredModuleStates, maxSpeed);
     desiredStateRobotRelative = config.toChassisSpeeds(desiredModuleStates);
 
     // Special case: desiredState is a complete stop. In this case, module angle is arbitrary, so
@@ -142,13 +143,13 @@ public class SwerveSetpointGenerator {
           prevSetpoint.moduleStates()[m].angle.getSin() * prevSetpoint.moduleStates()[m].velocity;
       prev_heading[m] = prevSetpoint.moduleStates()[m].angle;
       if (prevSetpoint.moduleStates()[m].velocity < 0.0) {
-        prev_heading[m] = prev_heading[m].rotateBy(Rotation2d.k180deg);
+        prev_heading[m] = prev_heading[m].rotateBy(Rotation2d.PI);
       }
       desired_vx[m] = desiredModuleStates[m].angle.getCos() * desiredModuleStates[m].velocity;
       desired_vy[m] = desiredModuleStates[m].angle.getSin() * desiredModuleStates[m].velocity;
       desired_heading[m] = desiredModuleStates[m].angle;
       if (desiredModuleStates[m].velocity < 0.0) {
-        desired_heading[m] = desired_heading[m].rotateBy(Rotation2d.k180deg);
+        desired_heading[m] = desired_heading[m].rotateBy(Rotation2d.PI);
       }
       if (all_modules_should_flip) {
         double required_rotation_rad =
@@ -209,7 +210,7 @@ public class SwerveSetpointGenerator {
                 .unaryMinus()
                 .rotateBy(desiredModuleStates[m].angle);
         if (flipHeading(necessaryRotation)) {
-          necessaryRotation = necessaryRotation.rotateBy(Rotation2d.kPi);
+          necessaryRotation = necessaryRotation.rotateBy(Rotation2d.PI);
         }
 
         // getRadians() bounds to +/- Pi.
@@ -278,7 +279,8 @@ public class SwerveSetpointGenerator {
       double reverseModuleTorque = config.moduleConfig.driveMotor.getTorque(reverseCurrentDraw);
 
       double prevSpeed = prevSetpoint.moduleStates()[m].velocity;
-      desiredModuleStates[m].optimize(prevSetpoint.moduleStates()[m].angle);
+      desiredModuleStates[m] =
+          desiredModuleStates[m].optimize(prevSetpoint.moduleStates()[m].angle);
       double desiredSpeed = desiredModuleStates[m].velocity;
 
       int forceSign;
@@ -292,7 +294,7 @@ public class SwerveSetpointGenerator {
         moduleTorque -= config.moduleConfig.torqueLoss;
         forceSign = 1; // Force will be applied in direction of module
         if (prevSpeed < 0) {
-          forceAngle = forceAngle.plus(Rotation2d.k180deg);
+          forceAngle = forceAngle.plus(Rotation2d.PI);
         }
       } else {
         moduleTorque = reverseModuleTorque;
@@ -300,7 +302,7 @@ public class SwerveSetpointGenerator {
         moduleTorque += config.moduleConfig.torqueLoss;
         forceSign = -1; // Force will be applied in opposite direction of module
         if (prevSpeed > 0) {
-          forceAngle = forceAngle.plus(Rotation2d.k180deg);
+          forceAngle = forceAngle.plus(Rotation2d.PI);
         }
       }
 
@@ -315,8 +317,8 @@ public class SwerveSetpointGenerator {
 
       // Calculate the torque this module will apply to the chassis
       if (!epsilonEquals(0, moduleForceVec.getNorm())) {
-        Rotation2d angleToModule = config.moduleLocations[m].getAngle();
-        Rotation2d theta = moduleForceVec.getAngle().minus(angleToModule);
+        Rotation2d angleToModule = config.moduleLocations[m].getAngle().orElse(Rotation2d.ZERO);
+        Rotation2d theta = moduleForceVec.getAngle().orElse(Rotation2d.ZERO).minus(angleToModule);
         chassisTorque += forceAtCarpet * config.modulePivotDistance[m] * theta.getSin();
       }
     }
@@ -390,7 +392,7 @@ public class SwerveSetpointGenerator {
       double wheelForceDist = wheelForces[m].getNorm();
       double appliedForce =
           wheelForceDist > 1e-6
-              ? wheelForceDist * wheelForces[m].getAngle().minus(retStates[m].angle).getCos()
+              ? wheelForceDist * wheelForces[m].getAngle().orElse(Rotation2d.ZERO).minus(retStates[m].angle).getCos()
               : 0.0;
       double wheelTorque = appliedForce * config.moduleConfig.wheelRadiusMeters;
       double torqueCurrent = config.moduleConfig.driveMotor.getCurrent(wheelTorque);
@@ -408,7 +410,7 @@ public class SwerveSetpointGenerator {
       final var deltaRotation =
           prevSetpoint.moduleStates()[m].angle.unaryMinus().rotateBy(retStates[m].angle);
       if (flipHeading(deltaRotation)) {
-        retStates[m].angle = retStates[m].angle.rotateBy(Rotation2d.k180deg);
+        retStates[m].angle = retStates[m].angle.rotateBy(Rotation2d.PI);
         retStates[m].velocity *= -1.0;
         appliedForce *= -1.0;
         torqueCurrent *= -1.0;
